main.py

```diff
@@ -154,22 +177,20 @@ def count_beer(travel_df):
     Count how many types of beer each brewery has
     """
     number_of_beers = 0
-    for row in travel_df.index:
-        br_id = travel_df.loc[row]["brewery_id"]
-        beers = beer_count_df.loc[br_id].values[0]
+    for row in travel_df.itertuples():
+        beers = beer_count_df.loc[row.brewery_id].values[0]
         number_of_beers += beers
     return number_of_beers
 
 
-def construct_google_map_path(travel_df):
+def construct_google_map_path(travel_df, geocodes_df, start_lat, start_lon):
     """
     Make a web string for google maps
     """
     web_str = "http://www.google.com/maps/dir/"
     web_str += f"{start_lat},{start_lon}/"
-    for row in travel_df.index:
-        br_id = travel_df.loc[row]["brewery_id"]
-        lat, lon = geocodes_df.loc[br_id, "latitude":"longitude"]
+    for row in travel_df.itertuples():
+        lat, lon = geocodes_df.loc[row.brewery_id, "latitude":"longitude"]
```

Comment:
Same as above.
You could have lat, lon in travel_df in advance.
The only thing you should need to calculate each step should be distance.

Reply:
Fixed